### ISSUE TITLE
[BUG] disable idle timeout

### ISSUE THREAD

**Issue by aaronliu0130** [SideStore/SideStore#756]
### Describe the bug

iPhone goes to sleep normally even with disable idle timeout on

### Instructions to reproduce

1. Enable disable idle timeout
2. Refresh or install app

### What version of SideStore are you using?

0.5.8

### Other info

on iOS 17

**Comment by mahee96** [comment]
what do you expect to happen? keep screen awake from start of install till the end of install, when staying in the install screen?

**Comment by aaronliu0130** [comment]
Is that not what it does?

**Comment by mahee96** [comment]
<img width="884" alt="Image" src="https://github.com/user-attachments/assets/ca1dde60-4db8-4927-90a6-407b6ffbfe36" />


I don't know what was the intent of the initial code, but current performOperations() is asynchronous and finishes quickly and the timeout hold is immediately cancelled. 

So if you want it to be for the full of the install, then this is a bug.

**Comment by mahee96** [comment]
Fixed this in https://github.com/SideStore/SideStore/commit/c3847276f7457bafd83c897e64483b416873a284

test with latest nightly and report back if it satisfies your needs. 

Expectations: 
From the point of start of install/refresh and till end of the install it should disable the idleTimeout and keeping screen alive.

**Comment by mahee96** [comment]
close this issue if it works as expected after the nightly testing

**Comment by aaronliu0130** [comment]
uh why did installing the nightly remove the listings of all my apps

**Comment by aaronliu0130** [comment]
and the screen still automatically locks

**Comment by mahee96** [comment]
Okay will keep this issue open and check it later

**Comment by aaronliu0130** [comment]
The nightly also broke the custom icon feature

**Comment by mahee96** [comment]
Yeah it is known issue and will be fixed later, it was caused when were fixing the shortcut issue…

But this is not a blocker because most people wanted background refresh and shortcut/siri refresh to work first…

to help, you can create a issue for the icon change not working so that it is tracked.

**Comment by haomuch** [comment]
I can’t believe this bug has been open for more than 1.5 years. 

Maybe we can simply keep the screen on when the app is running in the foreground if this feature is activated.

**Comment by mahee96** [comment]
this should be fixed by now in 0.6.4-xxxx latest nightly. 

if it is not create a new issue and link to this as reference.
